libr/io/p/io_r2k.c

```diff
@@ -14,248 +14,278 @@ typedef  struct _PPA {
 	HANDLE hSCManager;
 	HANDLE hService;
 	BOOL ret = FALSE;
 
-	hSCManager = OpenSCManagerA(NULL, NULL, SC_MANAGER_CREATE_SERVICE);
-	if (hSCManager)	{
-		hService = OpenServiceA(hSCManager, lpServiceName, SERVICE_START | DELETE | SERVICE_STOP);
+	hSCManager = OpenSCManagerA (NULL, NULL, SC_MANAGER_CREATE_SERVICE);
+	if (hSCManager) {
+		hService = OpenServiceA (hSCManager, lpServiceName, SERVICE_START | DELETE | SERVICE_STOP);
 		if (hService) {
-			DeleteService(hService);
-			CloseServiceHandle(hService);
+			DeleteService (hService);
+			CloseServiceHandle (hService);
 			ret = TRUE;
 		}
-		CloseServiceHandle(hSCManager);
+		CloseServiceHandle (hSCManager);
 	}
 	return ret;
 }
-BOOL StartStopService(LPCSTR  lpServiceName, BOOL bStop) {
+BOOL StartStopService(LPCSTR lpServiceName, BOOL bStop) {
 	HANDLE hSCManager;
 	HANDLE hService;
 	SERVICE_STATUS ssStatus;
 	BOOL ret = FALSE;
-	hSCManager = OpenSCManagerA(NULL, NULL, SC_MANAGER_CREATE_SERVICE);
+	hSCManager = OpenSCManagerA (NULL, NULL, SC_MANAGER_CREATE_SERVICE);
 	if (hSCManager)	{
-		hService = OpenServiceA(hSCManager, lpServiceName, SERVICE_START | DELETE | SERVICE_STOP);
+		hService = OpenServiceA (hSCManager, lpServiceName, SERVICE_START | DELETE | SERVICE_STOP);
 		if (hService) {
 			if (!bStop) {
-				if (StartServiceA(hService, 0, NULL)) {
-					eprintf("Service started [OK]\n");
+				if (StartServiceA (hService, 0, NULL)) {
+					eprintf ("Service started [OK]\n");
 					ret = TRUE;
 				}
 				else {
-					eprintf("Service started [FAIL]\n");
+					eprintf ("Service started [FAIL]\n");
 				}
 			}
 			else {
-				if (ControlService(hService, SERVICE_CONTROL_STOP, &ssStatus)) {
-					printf("Service Stopped [OK]\n");
+				if (ControlService (hService, SERVICE_CONTROL_STOP, &ssStatus)) {
+					printf ("Service Stopped [OK]\n");
 					ret = TRUE;
 				}
 				else {
-					printf("Service Stopped [FAIL]\n");
+					printf ("Service Stopped [FAIL]\n");
 				}
 			}
-			CloseServiceHandle(hService);
-			DeleteService(hService);
+			CloseServiceHandle (hService);
+			DeleteService (hService);
 		}
-		CloseServiceHandle(hSCManager);
+		CloseServiceHandle (hSCManager);
 	}
 	return ret;
 }
 BOOL InitDriver(VOID)
 {
-	BOOL	Ret = FALSE;
-	gHandleDriver = CreateFileA(strDeviceName, GENERIC_READ | GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_ALWAYS, FILE_ATTRIBUTE_DIRECTORY, 0);
+	gHandleDriver = CreateFileA (strDeviceName, GENERIC_READ | GENERIC_WRITE, FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_ALWAYS, FILE_ATTRIBUTE_DIRECTORY, 0);
 	if (gHandleDriver != INVALID_HANDLE_VALUE)
-		Ret = TRUE;
-	return(Ret);
+	{
+		return TRUE;
+	}
+	return FALSE;
 }
-char *GetFileName(const char *path)
-{
-	char *pfile;
-	pfile = path + strlen(path);
-	for (; pfile > path; pfile--)
-	{
-		if ((*pfile == '\\') || (*pfile == '/'))
-		{
-			pfile++;
-			break;
-		}
-	}
-	return pfile;
+char *GetFileName(unsigned char *path)
+{
+	char *pfile;
+	pfile = path + strlen (path);
```

Comment:
Theres a function in rutil named r_str_lchr that finds the last occurrence of a char. Use that instead of loop by hand